core/src/main/java/org/infinispan/configuration/cache/ClusteringConfiguration.java

```diff
@@ -136,6 +136,41 @@ public AttributeSet attributes() {
       return attributes;
    }
 
+   @Override
+   public boolean matches(ClusteringConfiguration other) {
+      if (attributes == null) {
+         if (other.attributes != null)
+            return false;
+      } else if (!attributes.matches(other.attributes))
+         return false;
+      if (hashConfiguration == null) {
```

Comment:
is it possible to the `*Configuration` to be `null`?

Reply:
I copied equals and s/equals/matches/ :)

Reply:
suggestion:
```
return hashConfiguration.matches(other.hashConfiguration) &&
          l1Configuration.matches(other.l1Configuration) &&
          ...
```

Reply:
ok